pkg/parser/ast/sem.go

```diff
@@ -0,0 +1,1315 @@
+	// DropDatabaseCommand represents DROP DATABASE statement
+	DropDatabaseCommand = "DROP DATABASE"
+	// DropIndexCommand represents DROP INDEX statement
+	DropIndexCommand = "DROP INDEX"
+	// DropPlacementPolicyCommand represents DROP PLACEMENT POLICY statement
+	DropPlacementPolicyCommand = "DROP PLACEMENT POLICY"
+	// DropResourceGroupCommand represents DROP RESOURCE GROUP statement
+	DropResourceGroupCommand = "DROP RESOURCE GROUP"
+	// DropSequenceCommand represents DROP SEQUENCE statement
+	DropSequenceCommand = "DROP SEQUENCE"
+	// DropTableCommand represents DROP TABLE statement
+	DropTableCommand = "DROP TABLE"
+	// DropViewCommand represents DROP VIEW statement
+	DropViewCommand = "DROP VIEW"
+	// DropUserCommand represents DROP USER statement
+	DropUserCommand = "DROP USER"
+	// FlashBackDatabaseCommand represents FLASHBACK DATABASE statement
+	FlashBackDatabaseCommand = "FLASHBACK DATABASE"
+	// FlashBackTableCommand represents FLASHBACK TABLE statement
+	FlashBackTableCommand = "FLASHBACK TABLE"
+	// FlashBackClusterCommand represents FLASHBACK CLUSTER statement
+	FlashBackClusterCommand = "FLASHBACK CLUSTER"
+	// LockTablesCommand represents LOCK TABLES statement
+	LockTablesCommand = "LOCK TABLES"
+	// OptimizeTableCommand represents OPTIMIZE TABLE statement
+	OptimizeTableCommand = "OPTIMIZE TABLE"
+	// RecoverTableCommand represents RECOVER TABLE statement
+	RecoverTableCommand = "RECOVER TABLE"
+	// RenameTableCommand represents RENAME TABLE statement
+	RenameTableCommand = "RENAME TABLE"
+	// RenameUserCommand represents RENAME USER statement
+	RenameUserCommand = "RENAME USER"
+	// AdminRepairTableCommand represents ADMIN REPAIR TABLE statement
+	AdminRepairTableCommand = "ADMIN REPAIR TABLE"
+	// TruncateTableCommand represents TRUNCATE TABLE statement
+	TruncateTableCommand = "TRUNCATE TABLE"
+	// UnlockTablesCommand represents UNLOCK TABLES statement
+	UnlockTablesCommand = "UNLOCK TABLES"
+)
+
+// DML Statements
+
+const (
+	// CallCommand represents CALL statement
+	CallCommand = "CALL"
+	// DeleteCommand represents DELETE statement
+	DeleteCommand = "DELETE"
+	// DistributeTableCommand represents DISTRIBUTE TABLE statement
+	DistributeTableCommand = "DISTRIBUTE TABLE"
+	// ImportIntoCommand represents IMPORT INTO statement
+	ImportIntoCommand = "IMPORT INTO"
+	// ReplaceCommand represents REPLACE statement
+	ReplaceCommand = "REPLACE"
+	// InsertCommand represents INSERT statement
+	InsertCommand = "INSERT"
+	// LoadDataCommand represents LOAD DATA statement
+	LoadDataCommand = "LOAD DATA"
+	// BatchCommand represents BATCH statement
+	BatchCommand = "BATCH"
+	// SelectCommand represents SELECT statement
+	SelectCommand = "SELECT"
+	// UnionCommand represents UNION statement
+	UnionCommand = "UNION"
+	// SplitRegionCommand represents SPLIT REGION statement
+	SplitRegionCommand = "SPLIT REGION"
+	// UpdateCommand represents UPDATE statement
+	UpdateCommand = "UPDATE"
+)
+
+// Show Statements
+
+const (
```

Comment:
I have some concerns about how can we maintain them...

For example, we support `SHOW PROFILE` in syntax only(it does nothing), so should we include it in this list?

Reply:
> For example, we support `SHOW PROFILE` in syntax only(it does nothing), so should we include it in this list?

I think we should include every syntax supported by the parser. This list is not exposed outside (to the user), so it's fine to have more commands than what TiDB actually supported.